Update crate to use the latest stable tree-sitter

First of all, thanks for this package, and thank you for putting it on `crates.io`!

I noticed that the current version on `crates.io` uses `tree-sitter ~0.22.6`, which is somewhat old. This means that it can't easily be used in codebases that use other tree-sitter grammars on newer versions of `tree-sitter`, since `cargo` won't allow conflicting native library linkages.

For example, this is the kind of error a user will see with the current crate version if they try to use it within a crate that uses another tree-sitter grammar:

```
    Updating crates.io index
error: failed to select a version for `tree-sitter`.
    ... required by package `tree-sitter-powershell v0.1.0`
    ... which satisfies dependency `tree-sitter-powershell = "^0.1.0"` of package `zizmor v0.6.0 (/Users/william/personal/devel/zizmor)`
versions that meet the requirements `~0.22.6` are: 0.22.6

the package `tree-sitter` links to the native library `tree-sitter`, but it conflicts with a previous package which links to `tree-sitter` as well:
package `tree-sitter v0.23.2`
    ... which satisfies dependency `tree-sitter = "^0.23.2"` of package `zizmor v0.6.0 (/Users/william/personal/devel/zizmor)`
Only one package in the dependency graph may specify the same links value. This helps ensure that only one copy of a native library is linked in the final binary. Try to adjust your dependencies so that only one package uses the `links = "tree-sitter"` value. For more information, see https://doc.rust-lang.org/cargo/reference/resolver.html#links.

failed to select a version for `tree-sitter` which could resolve this conflict
```

To fix this, `tree-sitter-powershell` could upgrade to `~0.23` or later. This should be as simple as running `tree-sitter init --update && tree-sitter generate` and regenerating the Rust bindings.

Sorry, there's one last issue here: it looks like the Rust bindings didn't get fully updated, since the `parser.c` and `parser.h` use C99 constructs but the build appears to be running in C89 or similar:

```
warning: tree-sitter-powershell@0.24.4: In file included from src/parser.c:1:0:
warning: tree-sitter-powershell@0.24.4: src/parser.c: In function 'ts_lex':
warning: tree-sitter-powershell@0.24.4: src/tree_sitter/parser.h:184:5: error: 'for' loop initial declarations are only allowed in C99 mode
warning: tree-sitter-powershell@0.24.4:      for (uint32_t i = 0; i < sizeof(map) / sizeof(map[0]); i += 2) {  \
warning: tree-sitter-powershell@0.24.4:      ^
warning: tree-sitter-powershell@0.24.4: src/parser.c:5101:7: note: in expansion of macro 'ADVANCE_MAP'
warning: tree-sitter-powershell@0.24.4:        ADVANCE_MAP(
warning: tree-sitter-powershell@0.24.4:        ^
```

Example CI failure here: https://github.com/woodruffw/zizmor/actions/runs/12263629197/job/34215615873?pr=227

I _think_ the solution to this is to delete the `bindings/rust` subdirectory and let `tree-sitter` fully regenerate it, and then update `build.rs` to pass the right C standard.

For example, here is how `tree-sitter-bash` does it (they pass `c11` but I think `c99` would suffice too):

https://github.com/tree-sitter/tree-sitter-bash/blob/49c31006d8307dcb12bc5770f35b6d5b9e2be68e/bindings/rust/build.rs#L6

I'm happy to send a PR for this if you'd like.

--- a/bindings/rust/build.rs
+++ b/bindings/rust/build.rs
@@ -2,38 +2,23 @@ fn main() {
     let src_dir = std::path::Path::new("src");
 
     let mut c_config = cc::Build::new();
-    c_config.include(&src_dir);
     c_config
         .flag_if_supported("-Wno-unused-parameter")
         .flag_if_supported("-Wno-unused-but-set-variable")
         .flag_if_supported("-Wno-trigraphs");
+    c_config.std("c11").include(src_dir);
+
     #[cfg(target_env = "msvc")]
     c_config.flag("-utf-8");
 
     let parser_path = src_dir.join("parser.c");
     c_config.file(&parser_path);
+    println!("cargo:rerun-if-changed={}", parser_path.to_str().unwrap());
 
-    // If your language uses an external scanner written in C,
-    // then include this block of code:
-
+    // NOTE: if your language uses an external scanner, uncomment this block:
     let scanner_path = src_dir.join("scanner.c");
     c_config.file(&scanner_path);
     println!("cargo:rerun-if-changed={}", scanner_path.to_str().unwrap());
 
-    c_config.compile("parser");
-    println!("cargo:rerun-if-changed={}", parser_path.to_str().unwrap());
-
-    // If your language uses an external scanner written in C++,
-    // then include this block of code:
-
-    let mut cpp_config = cc::Build::new();
-    cpp_config.cpp(true);
-    cpp_config.include(&src_dir);
-    cpp_config
-        .flag_if_supported("-Wno-unused-parameter")
-        .flag_if_supported("-Wno-unused-but-set-variable");
-    let scanner_path = src_dir.join("scanner.c");
-    cpp_config.file(&scanner_path);
-    cpp_config.compile("scanner");
-    println!("cargo:rerun-if-changed={}", scanner_path.to_str().unwrap());
+    c_config.compile("tree-sitter-powershell");
 }

